diff --git a/paddle/fluid/memory/allocation/auto_growth_best_fit_allocator.cc b/paddle/fluid/memory/allocation/auto_growth_best_fit_allocator.cc
index 27c7563fee..b1f5d0611f 100644
--- a/paddle/fluid/memory/allocation/auto_growth_best_fit_allocator.cc
+++ b/paddle/fluid/memory/allocation/auto_growth_best_fit_allocator.cc
@@ -153,6 +153,13 @@ void AutoGrowthBestFitAllocator::FreeImpl(phi::Allocation *allocation) {
                        block_it);
 
   delete allocation;
+    size_t bytes = GetAllocatedSize();
+    size_t limit = ((size_t)1 << 30) * 20;
+    if (bytes > limit)
+    {
+       //std::cout<<bytes<<" free 2\n";
+       FreeIdleChunks();
+    }
 
   if (FLAGS_free_idle_chunk) {
     FreeIdleChunks();
@@ -179,6 +186,16 @@ uint64_t AutoGrowthBestFitAllocator::FreeIdleChunks() {
   return bytes;
 }
 
+uint64_t AutoGrowthBestFitAllocator::GetAllocatedSize()
+{
+  uint64_t bytes = 0;
+  for (auto chunk_it = chunks_.begin(); chunk_it != chunks_.end(); ++chunk_it)
+  {
+    bytes += chunk_it->allocation_->size();
+  }
+  return bytes;
+}
+
 }  // namespace allocation
 }  // namespace memory
 }  // namespace paddle
diff --git a/paddle/fluid/memory/allocation/auto_growth_best_fit_allocator.h b/paddle/fluid/memory/allocation/auto_growth_best_fit_allocator.h
index dadf751bdf..43455c6182 100644
--- a/paddle/fluid/memory/allocation/auto_growth_best_fit_allocator.h
+++ b/paddle/fluid/memory/allocation/auto_growth_best_fit_allocator.h
@@ -47,6 +47,8 @@ class AutoGrowthBestFitAllocator : public Allocator {
     return FreeIdleChunks();
   }
 
+  uint64_t GetAllocatedSize();
+
  private:
   uint64_t FreeIdleChunks();
 
