--- a/cmake/hip.cmake
+++ b/cmake/hip.cmake
@@ -66,0 +67,4 @@
+list(APPEND HIP_HCC_FLAGS --amdgpu-target=gfx926)
+list(APPEND HIP_HCC_FLAGS --amdgpu-target=gfx928)
+list(APPEND HIP_HCC_FLAGS --amdgpu-target=gfx936)
+list(APPEND HIP_HCC_FLAGS --amdgpu-target=gfx938)
@@ -68,0 +74,4 @@
+list(APPEND HIP_CLANG_FLAGS --amdgpu-target=gfx926)
+list(APPEND HIP_CLANG_FLAGS --amdgpu-target=gfx928)
+list(APPEND HIP_CLANG_FLAGS --amdgpu-target=gfx936)
+list(APPEND HIP_CLANG_FLAGS --amdgpu-target=gfx938)
