diff --git a/gloo/common/linux.cc b/gloo/common/linux.cc
index a3726da..9a42a12 100644
--- a/linux.cc
+++ b/linux.cc
@@ -188,8 +188,8 @@ static int getInterfaceSpeedGLinkSettings(int sock, struct ifreq* ifr) {
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(4,6,0)
   constexpr auto link_mode_data_nwords = 3 * 127;
   struct {
-    struct ethtool_link_settings req;
     __u32 link_mode_data[link_mode_data_nwords];
+    struct ethtool_link_settings req;
   } ecmd;
   int rv;
