diff --git a/pocketfft_hdronly.h b/pocketfft_hdronly.h
index d75ada6..d019b56 100644
--- a/pocketfft_hdronly.h
+++ b/pocketfft_hdronly.h
@@ -153,7 +153,7 @@ template<> struct VLEN<double> { static constexpr size_t val=2; };
 inline void *aligned_alloc(size_t align, size_t size)
   {
   // aligned_alloc() requires that the requested size is a multiple of "align"
-  void *ptr = ::aligned_alloc(align,(size+align-1)&(~(align-1)));
+  void *ptr = std::aligned_alloc(align,(size+align-1)&(~(align-1)));
   if (!ptr) throw std::bad_alloc();
   return ptr;
   }
