diff --git a/include/pybind11/detail/internals.h b/include/pybind11/detail/internals.h
index 4ea37486..a4c72364 100644
--- a/include/pybind11/detail/internals.h
+++ b/include/pybind11/detail/internals.h
@@ -129,7 +129,14 @@ public:
     }
 
 private:
+#if defined(__GNUC__) && !defined(__clang__) && __GNUC__ <= 8
+#    pragma GCC diagnostic push
+#    pragma GCC diagnostic ignored "-Wmissing-field-initializers"
+#endif
     PYBIND11_TLS_KEY_INIT(mutable key_)
+#if defined(__GNUC__) && !defined(__clang__) && __GNUC__ <= 8
+#    pragma GCC diagnostic pop
+#endif
 };
 
 PYBIND11_NAMESPACE_BEGIN(detail)
