diff --git a/include/pybind11/detail/internals.h b/include/pybind11/detail/internals.h
index a49cdaa4..19fff38f 100644
--- a/include/pybind11/detail/internals.h
+++ b/include/pybind11/detail/internals.h
@@ -129,3 +129,10 @@
 private:
+#if defined(__GNUC__) && !defined(__clang__) && __GNUC__ <= 8
+#    pragma GCC diagnostic push
+#    pragma GCC diagnostic ignored "-Wmissing-field-initializers"
+#endif
     PYBIND11_TLS_KEY_INIT(mutable key_)
+#if defined(__GNUC__) && !defined(__clang__) && __GNUC__ <= 8
+#    pragma GCC diagnostic pop
+#endif
 };
