diff --git a/libcudacxx/include/cuda/std/detail/libcxx/include/cmath b/libcudacxx/include/cuda/std/detail/libcxx/include/cmath
index ab25d7585e..99f144f4b7 100644
--- a/libcudacxx/include/cuda/std/detail/libcxx/include/cmath
+++ b/libcudacxx/include/cuda/std/detail/libcxx/include/cmath
@@ -830,7 +830,11 @@ _LIBCUDACXX_CONSTEXPR_AFTER_CXX11_COMPLEX _Tp __constexpr_logb(_Tp __x) {
     return _Tp(__exp);
   }
 #endif // defined(_LIBCUDACXX_IS_CONSTANT_EVALUATED)
+#ifdef __XCN__
+  return ::logb(__x);
+#else
   return __builtin_logb(__x);
+#endif
 }
 #endif // !_MSVC
 
