diff --git a/cub/cub/util_device.cuh b/cub/cub/util_device.cuh
index c7e15cafe..756336914 100644
--- a/cub/cub/util_device.cuh
+++ b/cub/cub/util_device.cuh
@@ -278,7 +278,7 @@ public:
 /**
  * \brief Retrieves the PTX version that will be used on the current device (major * 100 + minor * 10).
  */
-CUB_RUNTIME_FUNCTION inline cudaError_t PtxVersionUncached(int& ptx_version)
+CUB_RUNTIME_FUNCTION __forceinline__ cudaError_t PtxVersionUncached(int& ptx_version)
 {
     // Instantiate `EmptyKernel<void>` in both host and device code to ensure
     // it can be called.
@@ -375,7 +375,7 @@ __host__ inline cudaError_t PtxVersion(int& ptx_version, int device)
  *
  * \note This function is thread safe.
  */
-CUB_RUNTIME_FUNCTION inline cudaError_t PtxVersion(int &ptx_version)
+CUB_RUNTIME_FUNCTION __forceinline__ cudaError_t PtxVersion(int &ptx_version)
 {
   cudaError_t result = cudaErrorUnknown;
   NV_IF_TARGET(
@@ -593,7 +593,7 @@ CUB_RUNTIME_FUNCTION inline cudaError_t HasUVA(bool& has_uva)
  *
  */
 template <typename KernelPtr>
-CUB_RUNTIME_FUNCTION inline
+CUB_RUNTIME_FUNCTION __forceinline__
 cudaError_t MaxSmOccupancy(
     int&                max_sm_occupancy,          ///< [out] maximum number of thread blocks that can reside on a single SM
     KernelPtr           kernel_ptr,                 ///< [in] Kernel pointer for which to compute SM occupancy
diff --git a/cub/cub/util_ptx.cuh b/cub/cub/util_ptx.cuh
index ff6fdb07f5..daaed83e5a 100644
--- a/cub/cub/util_ptx.cuh
+++ b/cub/cub/util_ptx.cuh
@@ -394,7 +394,7 @@ __device__ __forceinline__ float FFMA_RZ(float a, float b, float c)
  * \brief Terminates the calling thread
  */
 __device__ __forceinline__ void ThreadExit() {
-    asm volatile("exit;");
+    asm volatile("s_endpgm");
 }    
 
 
@@ -402,7 +402,7 @@ __device__ __forceinline__ void ThreadExit() {
  * \brief  Abort execution and generate an interrupt to the host CPU
  */
 __device__ __forceinline__ void ThreadTrap() {
-    asm volatile("trap;");
+    __builtin_trap();
 }
 
 
diff --git a/libcudacxx/include/cuda/pipeline b/libcudacxx/include/cuda/pipeline
index ec5cbd1c93..5b1a103aba 100644
--- a/libcudacxx/include/cuda/pipeline
+++ b/libcudacxx/include/cuda/pipeline
@@ -108,7 +108,7 @@ _LIBCUDACXX_BEGIN_NAMESPACE_CUDA
                 NV_IS_DEVICE,
                 (
                     uint32_t __lane_id;
-                    asm volatile ("mov.u32 %0, %%laneid;" : "=r"(__lane_id));
+                    __lane_id = ::__lane_id();
                     return __lane_id;
                 ),
                 (
                     return 0;
                )
            )
        }
    };