diff --git a/Eigen/src/Core/util/ConfigureVectorization.h b/Eigen/src/Core/util/ConfigureVectorization.h
index af4e69623..6944be650 100644
--- a/Eigen/src/Core/util/ConfigureVectorization.h
+++ b/Eigen/src/Core/util/ConfigureVectorization.h
@@ -470,6 +470,16 @@
   #include <hip/hip_fp16.h>
 #endif
 
+#ifdef EIGEN_MUSACC
+  #define EIGEN_VECTORIZE_GPU
+  #include <vector_types.h>
+  #define EIGEN_HAS_MUSA_FP16
+#endif
+
+#ifdef EIGEN_HAS_MUSA_FP16
+  #include <musa_runtime_api.h>
+  #include <musa_fp16.h>
+#endif
 
 /** \brief Namespace containing all symbols from the %Eigen library. */
 namespace Eigen {
